docs/monorepo.md

```diff
@@ -0,0 +1,79 @@
+## Jetpack Monorepo Overview
+
+Welcome to the Jetpack Monorepo! This document will give you some idea of the layout, and what is required for your project to fix in with our tooling.
+
+### Layout
+
+Projects are divided into WordPress plugins, Composer packages, and Gutenberg editor extensions.
+
+* WordPress plugins live in subdirectories of `projects/plugins/`. The directory name should probably match the WordPress plugin name, with a leading "jetpack-" removed if applicable.
+* Composer packages live in subdirectories of `projects/packages/`. The directory name should probably match the package name with the leading "Automattic/jetpack-" removed.
+* Editor extensions live in subdirectories of `projects/editor-extensions/`. The directory name should match the feature name (without a "jetpack/" prefix).
+
+Tooling that's applicable to the monorepo as a whole, including tooling for generically handling projects, lives in `tools/`.
+
+Documentation that's applicable to the monorepo as a whole lives in `docs/`.
+
+All GitHub Actions configuration for the monorepo, including CI, lives in `.github`. We should strive to make things here generic rather than specifc to any one project.
+
+* Actual actions live in `.github/actions/`. If it doesn't have a `action.yml` file, it shouldn't be in there.
+* Pattern matchers (not associated with an action), go in `.github/matchers/`.
+* Other files specific to actions, including scripts used with `run:`, go in `.github/files/`.
+
+### Compatibility
+
+All projects should be compatible with PHP versions WordPress supports. That's currently PHP 5.6 to 8.0.
+
+We use eslint and phpcs to lint JavaScript and PHP code. Projects should comply with the standards enforced by these tools.
```

Comment:
Should we be more specific about which coding standards projects should comply with?

Reply:
Maybe just a link to the [Coding Standards](https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#coding-standards) section of the dev environment doc would be good?

Reply:
That sounds like a good idea to me.